### PR TITLE
Add API routes to get runner registration token

### PR DESCRIPTION
**Pull request by lunny**
Replace #23761

### CODE REVIEW

File: routers/api/v1/admin/runners.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package admin
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/routers/api/v1/shared"
+)
+
+// https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-organization
+
+// GetRegistrationToken list an organization's actions secrets
```
Comment (denyskon) [inline routers/api/v1/admin/runners.go]: Copy-paste?

File: routers/api/v1/org/runners.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package org
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/routers/api/v1/shared"
+)
+
+// https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-organization
+
+// GetRegistrationToken list an organization's actions secrets
```
Comment (denyskon) [inline routers/api/v1/org/runners.go]: as above

File: routers/api/v1/repo/runners.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/routers/api/v1/shared"
+)
+
+// GetRegistrationToken list an organization's actions secrets
```
Comment (denyskon) [inline routers/api/v1/repo/runners.go]: as above

**Comment by denyskon** [comment]
@lunny CI failure seems related

**Comment by lunny** [comment]
> @lunny CI failure seems related

Fixed

File: routers/api/v1/admin/runners.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+// GetRegistrationToken returns the token to register global runners
+func GetRegistrationToken(ctx *context.APIContext) {
+	// swagger:operation GET /admin/runners/registration-token admin adminGetRunnerRegistrationToken
+	// ---
+	// summary: Get an global actions runner registration token
+	// produces:
+	// - application/json
+	// parameters:
+	// responses:
```
Comment (puni9869) [inline routers/api/v1/admin/runners.go:21]: Could you tag the Error 4XX if applicable?

File: routers/api/v1/org/runners.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	// summary: Get an organization's actions runner registration token
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: org
+	//   in: path
+	//   description: name of the organization
+	//   type: string
+	//   required: true
+	// responses:
```
Comment (puni9869) [inline routers/api/v1/org/runners.go:26]: Error 4XX if required?

**Comment by denyskon** [comment]
Removing backport as it's a new feature and we're in feature freeze

File: routers/api/v1/shared/runners.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+// RegistrationToken is response related to registeration token
+// swagger:response RegistrationToken
+type RegistrationToken struct {
+	Token string `json:"token"`
+}
+
+func GetRegistrationToken(ctx *context.APIContext, ownerID, repoID int64) {
+	token, err := actions_model.GetLastestRunnerToken(ctx, ownerID, repoID)
```
Comment (denyskon) [inline routers/api/v1/shared/runners.go]:
```suggestion
	token, err := actions_model.GetLatestRunnerToken(ctx, ownerID, repoID)
```

File: routers/api/v1/repo/runners.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+package repo
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/routers/api/v1/shared"
+)
+
+// GetRegistrationToken returns the token to register repo runners
+func GetRegistrationToken(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/runners/registration-token repo repoGetRunnerRegistrationToken
```
Comment (denyskon) [inline routers/api/v1/repo/runners.go]:
```suggestion
	// swagger:operation GET /repos/{owner}/{repo}/runners/registration-token repository repoGetRunnerRegistrationToken
```

File: routers/api/v1/api.go
Comment (denyskon) [inline routers/api/v1/api.go:1]: Repo runner missing?

**Review by denyskon — COMMENTED** [review]
Please fix the issues I pointed out, but everything else looks good :+1:

**Comment by denyskon** [comment]
@lunny Seems that you have to run `make generate-swagger` again, and also add the repo runner endpoint inside `api.go`

File: routers/api/v1/admin/runners.go (hunk 4 above)
Comment (lunny) [reply routers/api/v1/admin/runners.go:21]: There is no 4XX error.

File: routers/api/v1/org/runners.go (hunk 5 above)
Comment (lunny) [reply routers/api/v1/org/runners.go:26]: As above.

File: routers/api/v1/api.go [hunk 8]
```diff
@@ -958,10 +958,16 @@ func Routes() *web.Route {
 				Delete(bind(api.DeleteEmailOption{}), user.DeleteEmail)
 
 			// create or update a user's actions secrets
```
Comment (denyskon) [inline routers/api/v1/api.go]:
```suggestion
			// manage user-level actions features
```

**Review by denyskon — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Pull request merged**